src/lsp-server.ts

```diff
@@ -256,6 +256,13 @@ export class LspServer {
                     full: true,
                     range: true,
                 },
+                workspace: {
+                    fileOperations: {
+                        willRename: {
+                            filters: [{ pattern: { glob: '**/*.{ts,js}', matches: 'file' } }],
```

Comment:
do we also need:
jsx
tsx
mjs
mts
cjs
cts

.d.ts
.d.mts
.d.cts

in the globs?

Reply:
Just let me check if I accidentally didn't made up some of the above file extensions.

Reply:
I don't think `cts` is a thing.

And the `.d.*` variants would probably be covered by `.*`? Not sure of the glob behavior here.

Reply:
> And the .d.* variants would probably be covered by .*?

yes `filename.d.ts` will match `*.ts` glob

Reply:
https://www.typescriptlang.org/docs/handbook/esm-node.html

> In turn, TypeScript supports two new source file extensions: .mts and .cts. When TypeScript emits these to JavaScript files, it will emit them to .mjs and .cjs respectively.